From b4c06525d4d9eff7801f092168565f43082a6db3 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 16 Mar 2014 12:04:49 +0000
Subject: [PATCH 5/5] MythMusic: re-enable the change image type option in the
 albumart editor

This uses the MUSIC_TAG_CHANGEIMAGE myth protocol command to change the
image type.
---
 mythplugins/mythmusic/mythmusic/editmetadata.cpp |   66 +++-------------------
 1 file changed, 8 insertions(+), 58 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/editmetadata.cpp b/mythplugins/mythmusic/mythmusic/editmetadata.cpp
index 2618b81..67979ff 100644
--- a/mythplugins/mythmusic/mythmusic/editmetadata.cpp
+++ b/mythplugins/mythmusic/mythmusic/editmetadata.cpp
@@ -1078,7 +1078,7 @@ void EditAlbumartDialog::showMenu(void )
 
     if (m_coverartList->GetItemCurrent())
     {
-        //menu->AddButton(tr("Change Image Type"), NULL, true);
+        menu->AddButton(tr("Change Image Type"), NULL, true);
 
         if (GetMythDB()->GetNumSetting("AllowTagWriting", 0))
         {
@@ -1143,63 +1143,13 @@ void EditAlbumartDialog::customEvent(QEvent *event)
                     AlbumArtImage *image = qVariantValue<AlbumArtImage*> (item->GetData());
                     if (image)
                     {
-                        AlbumArtImage oldImage = *image;
-
-                        image->imageType = (ImageType) type;
-
-                        if (image->imageType == oldImage.imageType)
-                            return;
-
-                        // rename any cached image to match the new type
-                        if (image->embedded)
-                        {
-                            // update the new cached image filename
-                            image->filename = QString(GetConfDir() + "/MythMusic/AlbumArt/%1-%2.jpg")
-                                                .arg(m_metadata->ID())
-                                                .arg(AlbumArtImages::getTypeFilename(image->imageType));
-
-                            if (image->filename != oldImage.filename && QFile::exists(oldImage.filename))
-                            {
-                                // remove any old cached file with the same name as the new one
-                                QFile::remove(image->filename);
-                                // rename the old cached file to the new one
-                                QFile::rename(oldImage.filename, image->filename);
-
-                                // force the theme image cache to refresh the image
-                                GetMythUI()->RemoveFromCacheByFile(image->filename);
-                            }
-
-                            // change the image type in the tag if it supports it
-                            MetaIO *tagger = m_metadata->getTagger();
-
-                            if (tagger && tagger->supportsEmbeddedImages())
-                            {
-                                if (!tagger->changeImageType(m_metadata->Filename(), &oldImage, image->imageType))
-                                    LOG(VB_GENERAL, LOG_INFO, "EditAlbumartDialog: failed to change image type");
-                            }
-
-                            if (tagger)
-                                delete tagger;
-                        }
-                        else
-                        {
-                            QFileInfo fi(oldImage.filename);
-
-                            // get the new images filename
-                            image->filename = QString(fi.absolutePath() + "/%1.jpg")
-                                    .arg(AlbumArtImages::getTypeFilename(image->imageType));
-
-                            if (image->filename != oldImage.filename && QFile::exists(oldImage.filename))
-                            {
-                                // remove any old cached file with the same name as the new one
-                                QFile::remove(image->filename);
-                                // rename the old cached file to the new one
-                                QFile::rename(oldImage.filename, image->filename);
-
-                                // force the theme image cache to refresh the image
-                                GetMythUI()->RemoveFromCacheByFile(image->filename);
-                            }
-                        }
+                        QStringList strList("MUSIC_TAG_CHANGEIMAGE");
+                        strList << m_metadata->Hostname()
+                                << QString::number(m_metadata->ID())
+                                << QString::number(image->imageType)
+                                << QString::number(type);
+
+                        gCoreContext->SendReceiveStringList(strList);
 
                         m_albumArtChanged = true;
 
-- 
1.7.10.2

